utilities/DumpUnitary/Driver.cs

```diff
@@ -29,11 +29,18 @@ static void Main(string[] args)
                 for (int row = 0; row < size; ++row) {
                     // skip the first line of the file (header)
                     string line = fileContent[row + 1];
-                    // split the line into 3 space-separated elements: index, real part and complex part
+
+                    // The complex number that represents the amplitude is tab-separated from the rest of the fields, 
+                    // and the real and imaginary components within are separated with spaces.
                     string[] parts = line.Split('\t');
-                    // drop the index and store real and complex parts
-                    string real = parts[1];
-                    string complex = parts[2];
+                    // drop the index and store real and imaginary parts
+                    string amplitude = parts[1];
+                    string[] amplitudeParts = amplitude.Split(' ', StringSplitOptions.RemoveEmptyEntries);
+                    string real = amplitudeParts[0];
+                    string imagSign = amplitudeParts[1];
```

Comment:
I checked the format of the wavefunction dump on several more interesting cases, with positive and negative imaginary parts, and the negative imaginary parts are formatted as follows:

    ∣7❭:\t 0.000000 + -0.707107 i\t == \t***********          [ 0.500000 ]    ↓    [ -1.57080 rad ]

So `amplitudeParts[1]` is always just `+`, and the sign of the imaginary part is included in `amplitudeParts[2]`. You don't need `complex` variable, line 45 will be `unitary[row, column] = $"({real}, {imag})";` instead, and line 45 will use `imag` instead of `complex`.